bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/Utils.java

```diff
@@ -90,5 +92,25 @@ public String getElement() {
         }
     }
 
+    /**
+     * Given a {@link Page}, this method returns the correct URL, taking into account that the provided {@code page} might provide a
+     * vanity URL.
+     *
+     * @param request the current request, used to determine the server's context path
+     * @param page    the page
+     * @return the URL of the page identified by the provided {@code path}, or the original {@code path} if this doesn't identify a
+     * {@link Page}
+     */
+    public static <T> T getPolicyProperty(String property, T defaultValue, @NotNull SlingHttpServletRequest slingRequest) {
```

Comment:
Please remove this utils method as it is not required (see comment in PageImpl).